tests/testthat/test-clean-names.R

```diff
@@ -371,14 +371,36 @@ test_that("Tests for cases beyond default snake", {
   expect_warning(expect_equal(names(clean_names(test_df, "small_camel")), names(clean_names(test_df, "lower_camel"))))
 })
 
-test_that("errors if not called on a data.frame", {
+test_that("Tests for clean_names.default() on lists and vectors", {
+  test_v <- seq_along(testing_vector)
+  names(test_v) <- testing_vector
+  test_list <- as.list(test_v)
+  
+  # Warnings due to partially handled mu
+  clean_v <- suppressWarnings(clean_names(test_v))
```

Comment:
Very minor thing but suppressWarnings on a test on this & following line makes me nervous.  It looks like the way @billdenney has addressed this is with `expect_warning()` instead, does it work to swap in that function?  Your comment in the preceding line helps.  Then if we go back, we know that the warning behavior is intentional and permanent.

Reply:
Indeed, expect_warning is much cleaner, I should have started with that.